test/examples/ucp_client_server.c

```diff
@@ -332,6 +333,46 @@ static int parse_cmd(int argc, char *const argv[], char **server_addr)
     return 0;
 }
 
+static char* sockaddr_get_ip_str(struct sockaddr *sock_addr,
+                                 char *ip_str, size_t max_size)
+{
+    struct sockaddr_in  addr_in;
+    struct sockaddr_in6 addr_in6;
+
+    switch (sock_addr->sa_family) {
+    case AF_INET:
+        memcpy(&addr_in, sock_addr, sizeof(struct sockaddr_in));
+        inet_ntop(AF_INET, &addr_in.sin_addr, ip_str, max_size);
+        return ip_str;
+    case AF_INET6:
+        memcpy(&addr_in6, sock_addr, sizeof(struct sockaddr_in6));
+        inet_ntop(AF_INET6, &addr_in6.sin6_addr, ip_str, max_size);
+        return ip_str;
+    default:
+        return "Invalid address family";
+    }
+}
+
+static char* sockaddr_get_port_str(struct sockaddr *sock_addr,
+                                   char *port_str, size_t max_size)
+{
+    struct sockaddr_in  addr_in;
+    struct sockaddr_in6 addr_in6;
+
+    switch (sock_addr->sa_family) {
```

Comment:
@yosefe Is this what you meant?
i still get:
ucp_client_server.c:362: error: dereferencing pointer ‘({anonymous})’ does break strict-aliasing rules

Reply:
yes
maybe pass 'const sockaddr_storage *sock_addr' as input parameter would help?

Reply:
hmm that's what i had in the previous commits. the memcpy is supposed to fix it...

Reply:
mempy should fix in_addr/in_addr6 but not sa_family

Reply:
but the compilation error is on the sa_family check

Reply:
that's why i think passing sockaddr_storage should help